pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -1050,7 +1050,19 @@ public CompletableFuture<Void> closeAsync() {
             });
         }
 
-        return closeFuture;
+        return closeFuture.thenCompose(ignore -> {
+            if (retryLetterProducer != null){
+                return retryLetterProducer.closeAsync();
```

Comment:
Should we close them parallel? should we do some special exception handling here?